### PR TITLE
feat: use official aws-sdk instrumentation and emit SQS processing spans

### PR DESCRIPTION
**Pull request by harelmo-lumigo**
TL;DR: this PR moves the SQS spans to be produced by the official [AWS SDK instrumentation ](https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-aws-sdk#readme) rather than by our custom HTTP instrumentation.
By default, the new setup is behind a "feature flag" - `LUMIGO_USE_AWS_SDK_INSTRUMENTATION` - so we can test it with specific customers first.

Longer story:
The [aws-sdk instrumentation ](https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-aws-sdk#readme) is brought into our distro, with several points to note:
* Since the aws-sdk instrumentation takes over **all** AWS service calls done via the SDK, we will have spans emitted from both the instrumentation and our own HTTP instrumentation unless we mutually exclude them - this is done by adding a check in the http instrumentation for a global context key (env-var) indicating that an AWS SDK instrumentation is active.
* Since I did not want to replace all the AWS http-based instrumentation at once, I've started with the one that brigs a client-requested feature - emitting a "processing" span for each message received from SQS, tied to the "receive" span. The other AWS service calls will be still instrumented by the http instrumentation.
* The `aws-sdk` package is considered v2, and v3 is actually broken down to separate packages per service - e.g., `@aws-sdk/client-sqs`. The instrumentation supports both, but we selectively activate it by having two separate `Instrumentor` classes, one for each case
* The test for v2 and v3 are almost identical, with the exception of using a different test-app (due to the different API used by the two versions). A re-use might be possible, but might require some refactors to the existing testing infra that's out of scope for this PR.

Side dishes:
* Cleanup of an unused `requireIfAvailable` method and tests

### CODE REVIEW

**Review by GuyMoses — APPROVED** [review]

File: src/instrumentations/aws-sdk-v2/attribute-extractors.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            id: CommonUtils.getRandomString(10),
+            targetId: null,
+            triggeredBy: Triggers.MessageTrigger.SQS,
+            fromMessageIds: [messageId],
+            extra: { resource: span.attributes['messaging.url'] },
+          };
+
+          lumigoData = JSON.stringify({
+            trigger: [mainTrigger, ...Triggers.recursiveParseTriggers(inner, mainTrigger.id)],
+          });
```
[the remark this answers is not in the document]
Comment (harelmo-lumigo) [reply src/instrumentations/aws-sdk-v2/attribute-extractors.ts]: https://github.com/lumigo-io/opentelemetry-js-distro/blob/2c1dc5fd806abdc95ea941f4aeebc82085f414de/src/instrumentations/aws-sdk/attribute-extractors.test.ts#L40

File: src/instrumentations/aws-sdk-v2/attribute-extractors.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      const innerRaw = messages?.[0]?.Body ?? '';
+      if (innerRaw.search(Triggers.INNER_MESSAGES_IDENTIFIER_PATTERN) > 0) {
+        // TODO: what if the inner message is not a JSON?
+        const inner = safeJsonParse(innerRaw);
+        if (inner !== undefined) {
+          const mainTrigger = {
+            id: CommonUtils.getRandomString(10),
+            targetId: null,
+            triggeredBy: Triggers.MessageTrigger.SQS,
+            fromMessageIds: [messageId],
```
[the remark this answers is not in the document]
Comment (harelmo-lumigo) [reply src/instrumentations/aws-sdk-v2/attribute-extractors.ts]: I basically used the same logic that was applied in the HTTP instrumentation - https://github.com/lumigo-io/opentelemetry-js-distro/blob/542052f510408c3d45d9e3c3570dbad9b279948c/src/parsers/aws.ts#L210 - does it mean it's a bug there?

File: src/instrumentations/aws-sdk-v2/attribute-extractors.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+          });
+        }
+      }
+
+      return {
+        messageId,
+        lumigoData,
+        ...baseAttributes,
+      };
+    }
```
[the remark this answers is not in the document]
Comment (harelmo-lumigo) [reply src/instrumentations/aws-sdk-v2/attribute-extractors.ts]: Makes sense, but the original implementation used only the first one's id, so I kept mine aligned to it: https://github.com/lumigo-io/opentelemetry-js-distro/blob/542052f510408c3d45d9e3c3570dbad9b279948c/src/parsers/aws.ts#L200 
Was this actually a bug?

File: src/instrumentations/aws-sdk-v2/hooks.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  // Signal other instrumentations that an aws-sdk span is becoming inactive
+  setAwsInstrumentationSpanActive(false);
+};
+
+export const sqsProcessHook = (span: MutableSpan) => {
+  // @ts-expect-error - span kind is read-only
+  span.kind = SpanKind.INTERNAL;
+
+  delete span['attributes']['messaging.message_id'];
+};
```
[the remark this answers is not in the document]
Comment (harelmo-lumigo) [reply src/instrumentations/aws-sdk-v2/hooks.ts]: Actually, all I remember is that without removing it, we get an infinited loop in the transaction build but not sure why - can you please remind me why having both this and `messageId` on the span cause this?
